lib/ansible/galaxy/collection/__init__.py

```diff
@@ -166,13 +171,41 @@ def verify_local_collection(
     modified_content = []  # type: List[ModifiedContent]
 
     verify_local_only = remote_collection is None
-    if verify_local_only:
-        # partial away the local FS detail so we can just ask generically during validation
-        get_json_from_validation_source = functools.partial(_get_json_from_installed_dir, b_collection_path)
-        get_hash_from_validation_source = functools.partial(_get_file_hash, b_collection_path)
 
+    # partial away the local FS detail so we can just ask generically during validation
+    get_json_from_validation_source = functools.partial(_get_json_from_installed_dir, b_collection_path)
+    get_hash_from_validation_source = functools.partial(_get_file_hash, b_collection_path)
+
+    if not verify_local_only:
+        # Compare installed version versus requirement version
+        if local_collection.ver != remote_collection.ver:
+            err = (
+                "{local_fqcn!s} has the version '{local_ver!s}' but "
+                "is being compared to '{remote_ver!s}'".format(
+                    local_fqcn=local_collection.fqcn,
+                    local_ver=local_collection.ver,
+                    remote_ver=remote_collection.ver,
+                )
+            )
+            display.display(err)
+            result.success = False
+            return result
+
+    if verify_local_only:
         # since we're not downloading this, just seed it with the value from disk
         manifest_hash = get_hash_from_validation_source(MANIFEST_FILENAME)
+    elif remote_collection.signatures:
+        valid = True
+        b_manifest_file = os.path.join(b_collection_path, b'MANIFEST.json')
+        for signature in remote_collection.signatures:
+            b_detached_signature = to_bytes(signature, errors='surrogate_or_strict')
+            if not verify_file_signature(b_manifest_file, b_detached_signature, keyring):
+                valid = False
```

Comment:
Any reason not to interrupt the loop early?
```suggestion
                result.success = False
                return result
```

Reply:
I thought it was maybe more useful to show all the issues to the user since they can provide multiple signatures. I always prefer to see aggregate errors rather than play error whack-a-mole to know the extent of what's wrong.

Updated `result.success = False` and then used that outside of the loop instead of having the additional `valid` variable though.

Reply:
Makes sense.